conformance/utils/http/http.go

```diff
@@ -72,10 +80,37 @@ func MakeRequestAndExpectResponse(t *testing.T, r roundtripper.RoundTripper, gwA
 			req.Headers[name] = []string{value}
 		}
 	}
-	cReq, cRes, err := r.CaptureRoundTrip(req)
-	require.NoErrorf(t, err, "error making request")
 
-	ExpectResponse(t, cReq, cRes, expected)
+	// We expect eventual consistency once the resources are ready, so wait until
+	// we've made a successful request and gotten a response with desired status
+	// before making assertions on the response body.
+	require.Eventually(t, func() bool {
+		var err error
+
+		_, cRes, err := r.CaptureRoundTrip(req)
+		if err != nil {
+			t.Logf("Request failed, not ready yet: %v", err.Error())
+			return false
+		}
+
+		if cRes.StatusCode != expected.StatusCode {
+			t.Logf("Expected response to have status %d but got %d, not ready yet", expected.StatusCode, cRes.StatusCode)
+			return false
+		}
+
+		return true
+	}, maxConsistencyPeriodPerRequest, 1*time.Second, "error making request, never got expected status")
+
+	// Once we've made a successful request and gotten a response with the
```

Comment:
Sorry to nitpick this a lot... a bit worried about the logic here. Typically what I have seen is that config changes will go from "all old config" to "mix of old and new" before arriving at "new config". So if we assert that once we see the 'new' state its immediately always 'new', we may see issues.

For example, if we have 2 pods implementing the gateway and 1 gets the new config first.

Here is how the Istio logic for this works: https://github.com/istio/istio/blob/30a866fdb5ad65eccde3dafe21d19f455addbd72/pkg/test/util/retry/retry.go#L182.

Basically we test that we get a success N times in a row -- but if we fail part way through the check we continue on retrying.

Also a bit of nitpick, but I am not sure we need the `consistency_check_%d` to show up in the test result? Doing the above would make that pretty hard to retain anyways

Reply:
I agree with the logic suggested here by @howardjohn. Maybe a fairly low timeout like 30 seconds to get 3 consecutive successful requests would work?

Reply:
👍 Not nitpicking IMO @howardjohn ! This makes a lot of sense and will help us avoid similar issues when we make the number of pods implementing the gateway configurable beyond `1` in the near future.

Reply:
@howardjohn I've pushed some updates to handle intermittent success in the style of what you linked. Let me know what ya think!

You can see an example run w/ log output under **Run tests** [here](https://github.com/hashicorp/consul-api-gateway/runs/5793540213?check_suite_focus=true).